test/e2e/smoke/emptydir-workflow.yaml

```diff
@@ -0,0 +1,29 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: volumes-emptydir-
+  labels:
+    argo-e2e: true
+spec:
+  entrypoint: main
+
+  templates:
+    - name: main
+      container:
+        image: argoproj/argosay:v2
+        args:
+          - echo
+          - "hello emptyDir"
+          - /mnt/work/message
+        volumeMounts:
```

Comment:
@jessesuen why don't we just automatically mount empty dir for the path of each input and output?